utils/tests/test_hosts.py

```diff
@@ -0,0 +1,162 @@
+        cls.HostLocalClass = HostLocal
+
+    def test__init__(self):
+        """
+        Tests that the __init__ from the setup_class method worked correctly.
+        """
+        assert self.TestUser1.env == MilMoveEnv.LOCAL
+        assert self.TestUser1.host == "https://primelocal:9443"
+        assert re.search("static/tls/devlocal-mtls.cer", self.TestUser1.cert_kwargs["cert"][0])
+        assert re.search("static/tls/devlocal-mtls.key", self.TestUser1.cert_kwargs["cert"][1])
+
+    def test_set_milmove_env(self):
+        self.HostLocalClass.env = None
+        self.TestUser1.set_milmove_env("exp")
+
+        assert self.HostLocalClass.env == MilMoveEnv.EXP
+        assert self.TestUser1.env == MilMoveEnv.EXP
+        assert self.TestUser2.env == MilMoveEnv.EXP
+
+    def test_invalid_set_milmove_env(self):
+        self.HostLocalClass.env = None
+
+        with pytest.raises(ImplementationError):
+            self.TestUser1.set_milmove_env("test")
+
+    def test_set_host_name(self):
+        """
+        Tests the host name for the class can be set using the class env
+        and be called from a class instance
+        """
+        # reset all initally set values in class
+        self.HostLocalClass.host = None
+        self.HostLocalClass.env = MilMoveEnv.EXP
+
+        # host is None due to setting the class host to None above
+        assert self.TestUser1.host is None
+        assert self.TestUser2.host is None
+
+        self.TestUser1.set_host_name()
+        assert self.TestUser1.host == "https://api.exp.move.mil"
+        assert self.TestUser2.host == "https://api.exp.move.mil"
+
+    @mock.patch.dict(os.environ, {"MOVE_MIL_EXP_TLS_CERT": "test_cert", "MOVE_MIL_EXP_TLS_KEY": "test_key"})
+    def test_set_cert_kwargs(self):
+        # Test if the environment is experimental
+        self.HostLocalClass.cert_kwargs = None
+        self.TestUser1.set_cert_kwargs()
+
+        assert re.search("tls/exp_tls_cert_key.pem", self.TestUser1.cert_kwargs["cert"])
+        assert re.search("tls/dod-ca-60-61-bundle.pem", self.TestUser1.cert_kwargs["verify"])
+        assert re.search("tls/exp_tls_cert_key.pem", self.TestUser2.cert_kwargs["cert"])
+        assert re.search("tls/dod-ca-60-61-bundle.pem", self.TestUser2.cert_kwargs["verify"])
+
+        # Test if the environment is local
+        self.HostLocalClass.cert_kwargs = None
+        self.HostLocalClass.env = MilMoveEnv.LOCAL
+        self.TestUser1.set_cert_kwargs()
+
+        assert re.search("tls/devlocal-mtls.cer", self.TestUser1.cert_kwargs["cert"][0])
+        assert re.search("tls/devlocal-mtls.key", self.TestUser1.cert_kwargs["cert"][1])
+        assert re.search("tls/devlocal-mtls.cer", self.TestUser2.cert_kwargs["cert"][0])
+        assert re.search("tls/devlocal-mtls.key", self.TestUser2.cert_kwargs["cert"][1])
+
+    @mock.patch.dict(os.environ, {"MOVE_MIL_EXP_TLS_CERT": "test_cert", "MOVE_MIL_EXP_TLS_KEY": "test_key"})
+    def test_create_deokoyed_cert_file(self):
```

Comment:
I think this should be "deployed" instead of "deokoyed" 😄

Reply:
LOL